### PR TITLE
Features/task 4

### PR DESCRIPTION
**Pull request by pavelf07**
App is rewritten to typescript.

### CODE REVIEW

File: onboarding-task/src/reducers/itemOrder.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { OrderedSet } from 'immutable';
+import { ADD_ITEM, DELETE_ITEM } from '../actions/actionTypes';
+import { IAction } from '../actions/IAction';
+
+const itemOrder = (state = OrderedSet(), action: IAction) => {
```
Comment (PetrSvirak) [inline onboarding-task/src/reducers/itemOrder.ts]: It should be plural, shouldn't it?

File: onboarding-task/src/selectors/getViewItems.ts [hunk 2]
```diff
@@ -1,13 +0,0 @@
-const memoize = require('memoizee');
```
Comment (PetrSvirak) [inline onboarding-task/src/selectors/getViewItems.ts]: You lost file history. Be more careful on renaming, it can sometimes take an extra commit, however, with longer and longer history, it worths it.

File: onboarding-task/src/containers/Item.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+const { connect } = require('react-redux');
```
Comment (Suzii) [inline onboarding-task/src/containers/Item.ts]: And do not use require at all, have a look at the updated code conventions on how to import defaults in typescript :)

File: onboarding-task/src/containers/List.ts [hunk 4]
```diff
@@ -1,24 +1,22 @@
...
 
-const mapDispatchToProps = (dispatch: Dispatch<IAction>) => {
+const mapDispatchToProps = (dispatch: Dispatch) => {
   return {
     addItem: (text: string) => dispatch(addItem(text)),
   };
 };
 
-const ListContainer: React.ComponentClass<undefined> = connect(
+const ListContainer = connect(
```
Comment (Suzii) [inline onboarding-task/src/containers/List.ts]: Why did you delete `React.ComponentClass<undefined>`? Like this, any props could be sent to container and will be ignored.
